### PR TITLE
WIP: CGC/Decree support

### PR DESCRIPTION
**Pull request by atlas0fd00m**
we've allowed the ELF parser to handle "CGC" binaries by magic value...  but now we add support for the DECREE platform, with it's own analysis modules and workspace emulator.  the design may not be final (right now i'm using i_int() in the emulator for things which probably belong in an analysis module) but i wanted to share and ask for input.

### CODE REVIEW

File: vivisect/parsers/elf.py [hunk 1]
```diff
@@ -84,18 +84,20 @@ def makeRelocTable(vw, va, maxva, addbase, baseaddr):
     'amd64':'sysvamd64call',
 }
 
-def loadElfIntoWorkspace(vw, elf, filename=None):
+def loadElfIntoWorkspace(vw, elf, filename=None, arch=None, platform=None, filefmt='elf'):
```
Comment (invisig0th) [inline vivisect/parsers/elf.py]: these parameters seem like something the elf object should contain/identify ( and potentially allow you to override ).

File: vivisect/parsers/elf.py [hunk 2]
```diff
@@ -220,7 +224,7 @@ def loadElfIntoWorkspace(vw, elf, filename=None):
             makeRelocTable(vw, sva, sva+size, addbase, baseaddr)
 
         if sec.sh_flags & Elf.SHF_STRINGS:
-            print "FIXME HANDLE SHF STRINGS"
+            makeStringTable(vw, sva, sva+size)
```
Comment (invisig0th) [inline vivisect/parsers/elf.py]: haha, thanks :)

File: vivisect/parsers/elf.py (hunk 1 above)
Comment (atlas0fd00m) [inline vivisect/parsers/elf.py]: the cgc-specific stuff, and the 'elf' name never makes it to the Elf, they're acted upon in loadElfIntoWorkspace().  are you suggesting we make the Elf.Elf class identify these things, and have the loadElfIntoWorkspace() function draw them out through accessors?

**Comment by invisig0th** [comment]
Digging this delta, had ignored it to ensure avoidance of conflict of interest issues.  keep on rawking and we'll mainline this once it's high-and-tight. :)

**Comment by atlas0fd00m** [comment]
i don't see any purpose for this going forward. killing this pr, but not the branch.

**Pull request closed without merging**